test/solve.jl

```diff
@@ -6,7 +6,7 @@ using StaticArrays, Base.Test
                 elty in (Float64, Int)
 
         A = elty.(rand(-99:2:99,n,n))
-        b = A*ones(elty,n)
+        b = 2*A*ones(elty,n)
```

Comment:
Oof, this wasn't the best test. I'd frankly still be a little worried about mistakes in indexing into `b` in the various `solve` implementations. Perhaps make it so that the elements are all different?